tutorials/video_classification_example/finetune.py

```diff
@@ -0,0 +1,255 @@
+from pathlib import Path
+from argparse import Namespace
+from torchvision.transforms._transforms_video import CenterCropVideo
+from pytorchvideo.data import LabeledVideoDataset
+from pytorchvideo.data.clip_sampling import UniformClipSampler
+import pytorch_lightning as pl
+import torch
+from pytorchvideo.models.head import create_res_basic_head
+from torch import nn
+from torch.optim import Adam
+
+# HACK
+from train import *
+
+
+class UCF11DataModule(KineticsDataModule):
+
+    def __init__(
+        self,
+        root="./",
+        batch_size=32,
+        num_workers=8,
+        holdout_scene=None,
+        side_size = 256,
+        crop_size = 256,
+        clip_mean = (0.45, 0.45, 0.45),
+        clip_std = (0.225, 0.225, 0.225),
+        num_frames = 8,
+        sampling_rate = 8,
+        frames_per_second = 30
+    ):
+        super().__init__(Namespace(data_type='video', batch_size=batch_size, workers=num_workers))
+
+        self.root = Path(root) / 'action_youtube_naudio'
+        assert self.root.exists(), "Dataset not found."
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.holdout_scene = holdout_scene
+        self.side_size = side_size
+        self.mean = clip_mean
+        self.std = clip_std
+        self.crop_size = crop_size
+        self.num_frames = num_frames
+        self.sampling_rate = sampling_rate
+        self.frames_per_second = frames_per_second
+        self.clip_duration = (self.num_frames * self.sampling_rate) / self.frames_per_second
+
+        self.classes = [x.name for x in self.root.glob("*") if x.is_dir()]
+        self.id_to_label = dict(zip(range(len(self.classes)), self.classes))
+        self.class_to_label = dict(zip(self.classes, range(len(self.classes))))
+        self.num_classes = len(self.classes)
+
+
+        # TODO - too many repeated .glob calls here.
+        self.train_paths = []
+        self.val_paths = []
+        self.holdout_scenes = {}
+        for c in self.classes:
+
+            # Scenes within each class directory
+            scene_names = sorted(set(x.name for x in (self.root / c).glob("*") if x.is_dir() and x.name != 'Annotation'))
+            
+            # Holdout the last scene
+            # TODO - wrap this in a function so users can override the split logic
+            holdout_scene = scene_names[-1]
```

Comment:
Why do we only hold out the last scene? Wouldn't it be better to have a 20/80 val/train split?

Reply:
In the [paper](https://www.crcv.ucf.edu/papers/cvpr2009_liu1.pdf), I saw:

>  In the training phase, leave one out cross validation (LOOCV)
scheme is used.

I interpreted that as "pick a random scene from each class and hold it out". Please correct me if I'm wrong!!

Reply:
I updated to use 80% of scenes for train, 20% of scenes for val. Note scenes are folders containing videos w/ same scene and/or actor. We dont want to mix videos from same scene folder in train/val splits.